### PR TITLE
[MLA-1009] observable performance tests

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Simple performance tests. Tests stepping Agents that use CollectObservations and ObservableAttributes, and additional tracking on specific markers (so we can measure just sensor setup and stepping).

TBD - what's the right place for these to live, and should we run them in Yamato?

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://docs.unity3d.com/Packages/com.unity.test-framework.performance@2.2/manual/index.html
https://docs.google.com/document/d/1wdNopiwvINV5uFRORjOduebUeGQwWLSS1vchaNF7nnE/edit# (internal)

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Tests/Editor/Unity.ML-Agents.Editor.Tests.asmdef [hunk 1]
```diff
@@ -4,7 +4,8 @@
         "Unity.ML-Agents.Editor",
         "Unity.ML-Agents",
         "Unity.Barracuda",
-        "Unity.ML-Agents.CommunicatorObjects"
+        "Unity.ML-Agents.CommunicatorObjects",
+        "Unity.PerformanceTesting"
```
Comment (surfnerd) [inline com.unity.ml-agents/Tests/Editor/Unity.ML-Agents.Editor.Tests.asmdef:8]: we don't want these here, correct?

File: com.unity.ml-agents/Tests/Runtime/Unity.ML-Agents.Runtime.Tests.asmdef [hunk 2]
```diff
@@ -4,7 +4,8 @@
         "Unity.ML-Agents",
         "Unity.Barracuda",
         "Unity.ML-Agents.CommunicatorObjects",
-        "Unity.ML-Agents.Editor"
+        "Unity.ML-Agents.Editor",
+        "Unity.PerformanceTesting"
```
Comment (surfnerd) [inline com.unity.ml-agents/Tests/Runtime/Unity.ML-Agents.Runtime.Tests.asmdef:8]: same for this

File: com.unity.ml-agents/Tests/Editor/Unity.ML-Agents.Editor.Tests.asmdef (hunk 1 above)
Comment (surfnerd) [inline com.unity.ml-agents/Tests/Editor/Unity.ML-Agents.Editor.Tests.asmdef:8]:
```suggestion
        "Unity.ML-Agents.CommunicatorObjects"
```

File: com.unity.ml-agents/Tests/Runtime/Unity.ML-Agents.Runtime.Tests.asmdef (hunk 2 above)
Comment (surfnerd) [inline com.unity.ml-agents/Tests/Runtime/Unity.ML-Agents.Runtime.Tests.asmdef:8]:
```suggestion
        "Unity.ML-Agents.Editor"
```

File: .yamato/com.unity.ml-agents-performance.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    UNITY_VERSION: {{ editor.version }}
+  commands:
+    - python -m pip install unity-downloader-cli --extra-index-url https://artifactory.eu-cph-1.unityops.net/api/pypi/common-python/simple
+    - unity-downloader-cli -u {{ editor.version }} -c editor --wait --fast
+    - curl -s https://artifactory.internal.unity3d.com/core-automation/tools/utr-standalone/utr --output utr
+    - chmod +x ./utr
+    - ./utr --suite=editor --platform=StandaloneOSX --editor-location=.Editor --testproject=DevProject --artifacts_path=build/test-results --report-performance-data --performance-project-id=com.unity.ml-agents --zero-tests-are-ok=1
+  triggers:
+    cancel_old_ci: true
+    expression: |
```
Comment (surfnerd) [inline .yamato/com.unity.ml-agents-performance.yml:22]: do we want to run these for every commit, or nightly?

Comment (chriselion) [reply .yamato/com.unity.ml-agents-performance.yml:22]: TBD - maybe similar to the training tests, and run Mac on each PR, and all platforms nightly?

File: com.unity.ml-agents/Tests/Editor/Unity.ML-Agents.Editor.Tests.asmdef (hunk 1 above)
Comment (chriselion) [reply com.unity.ml-agents/Tests/Editor/Unity.ML-Agents.Editor.Tests.asmdef]: Reverted

File: com.unity.ml-agents/Tests/Runtime/Unity.ML-Agents.Runtime.Tests.asmdef (hunk 2 above)
Comment (chriselion) [reply com.unity.ml-agents/Tests/Runtime/Unity.ML-Agents.Runtime.Tests.asmdef]: Reverted

**Pull request merged**